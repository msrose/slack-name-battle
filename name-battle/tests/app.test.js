const axios = require('axios')
const MockAdapter = require('axios-mock-adapter')
const { URL } = require('url')

const mock = new MockAdapter(axios)

mock.onGet(/https:\/\/slack.com\/api\/users.info/).reply(config => [
    200,
    {
        ok: true,
        user: {
            profile: {
                real_name: new URL(config.url).searchParams.get('user'),
            },
        },
    },
])

const { lambdaHandler } = require('../app')

describe('Slack name battle', () => {
    it.each([
        ['target', 'attacker'],
        ['michael', 'michael'],
        ['aaaaa', 'a'],
        ['aaaaa', 'aa'],
    ])('conducts a name battle between %s and %s', async (target, attacker) => {
        const result = await lambdaHandler({
            body: `text=${target}&user_id=${attacker}`,
        })
        expect(result.statusCode).toBe(200)
        expect(JSON.parse(result.body)).toMatchSnapshot()
    })
})
