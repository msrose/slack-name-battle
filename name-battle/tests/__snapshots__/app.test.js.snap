// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack name battle conducts a name battle between aaaaa and a 1`] = `
Object {
  "response_type": "in_channel",
  "text": "<@a> challenges <@aaaaa> to a Name Battle! :collision: *_a_* attacks *_aaaaa_* in a parallel universe...
aaaaa is barely affected, with *80.00%* life force remaining! A paltry attack! :joy:",
}
`;

exports[`Slack name battle conducts a name battle between aaaaa and aa 1`] = `
Object {
  "response_type": "in_channel",
  "text": "<@aa> challenges <@aaaaa> to a Name Battle! :collision: *_aa_* attacks *_aaaaa_* in a parallel universe...
aaaaa is wounded, with *60.00%* life force remaining! A worrying attack! :gun:",
}
`;

exports[`Slack name battle conducts a name battle between michael and michael 1`] = `
Object {
  "response_type": "in_channel",
  "text": "<@michael> challenges <@michael> to a Name Battle! :collision: *_michael_* attacks *_michael_* in a parallel universe...
michael is dead, with *0%* life force remaining! A devastating attack! :skull:",
}
`;

exports[`Slack name battle conducts a name battle between target and attacker 1`] = `
Object {
  "response_type": "in_channel",
  "text": "<@attacker> challenges <@target> to a Name Battle! :collision: *_attacker_* attacks *_target_* in a parallel universe...
target is critically injured, with *16.67%* life force remaining! A destructive attack! :face_with_head_bandage:",
}
`;
